op1._ yields: Compiler Internal Error: Qualified names should not be present at codegen time

### Discord username

_No response_

### What type of issue is this?

Intermittent – Occurring irregularly

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

When parsing columns in the table on the third node with `parse` the compiler error occurred:
```router.ts:62 [ERROR] app/gui/src/model/project/synchronized.rs:591 Execution failed in context ca4e82f0-e987-4ba2-9dce-d6f9265d37af. Error: Execution of function main failed (Compiler Internal Error: Qualified names should not be present at codegen time.)..```
and the node was missing values:
<img width="710" alt="Zrzut ekranu 2023-08-15 o 08 10 06" src="https://github.com/enso-org/enso/assets/12892578/6dd9b8a5-bb9c-4071-abf6-a5fc06e5d77e">
After some time it recovers and it was possible to choose the proper column from the drop-down.

### Expected behaviour

The drop-down should work after appearing on the node.

### How we can reproduce it?

_No response_

### Screenshots or screencasts

_No response_

### Logs

[20230815-055256-813-enso-project-manager.log](https://github.com/enso-org/enso/files/12341657/20230815-055256-813-enso-project-manager.log)


### Enso Version

nightly 15.08

### Browser or standalone distribution

Standalone distribution (local project)

### Browser Version or standalone distribution

standalone

### Operating System

MacOS

### Operating System Version

_No response_

### Hardware you are using

_No response_

--- a/engine/runtime/src/main/java/org/enso/interpreter/node/MethodRootNode.java
+++ b/engine/runtime/src/main/java/org/enso/interpreter/node/MethodRootNode.java
@@ -1,16 +1,22 @@
 package org.enso.interpreter.node;
 
+import java.util.function.Supplier;
+
+import org.enso.compiler.core.CompilerError;
+import org.enso.interpreter.EnsoLanguage;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.data.Type;
+import org.enso.interpreter.runtime.data.text.Text;
+import org.enso.interpreter.runtime.error.PanicException;
+import org.enso.interpreter.runtime.scope.LocalScope;
+import org.enso.interpreter.runtime.scope.ModuleScope;
+
 import com.oracle.truffle.api.CompilerDirectives;
 import com.oracle.truffle.api.dsl.ReportPolymorphism;
 import com.oracle.truffle.api.frame.VirtualFrame;
 import com.oracle.truffle.api.nodes.Node;
 import com.oracle.truffle.api.nodes.NodeInfo;
 import com.oracle.truffle.api.source.SourceSection;
-import java.util.function.Supplier;
-import org.enso.interpreter.EnsoLanguage;
-import org.enso.interpreter.runtime.data.Type;
-import org.enso.interpreter.runtime.scope.LocalScope;
-import org.enso.interpreter.runtime.scope.ModuleScope;
 
 @ReportPolymorphism
 @NodeInfo(shortName = "Method", description = "A root node for Enso methods.")
@@ -142,9 +148,16 @@ public class MethodRootNode extends ClosureRootNode {
 
     @CompilerDirectives.TruffleBoundary
     final ExpressionNode replaceItself() {
+      try {
         ExpressionNode newNode = replace(provider.get());
         notifyInserted(newNode);
         return newNode;
+      } catch (CompilerError abnormalException) {
+        var ctx = EnsoContext.get(this);
+        var msg = Text.create(abnormalException.getMessage());
+        var load = ctx.getBuiltins().error().makeCompileError(msg);
+        throw new PanicException(load, this);
+      }
     }
   }
   public boolean isSubjectToInstrumentation() {

--- a/engine/runtime/src/test/java/org/enso/compiler/ExecCompilerTest.java
+++ b/engine/runtime/src/test/java/org/enso/compiler/ExecCompilerTest.java
@@ -2,6 +2,7 @@ package org.enso.compiler;
 
 import java.io.OutputStream;
 import java.nio.file.Paths;
+
 import org.enso.polyglot.RuntimeOptions;
 import org.graalvm.polyglot.Context;
 import org.graalvm.polyglot.PolyglotException;
@@ -99,6 +100,24 @@ public class ExecCompilerTest {
     assertTrue("We get an error value back", error.isException());
     assertTrue("The error value also represents null", error.isNull());
     assertEquals("(Error: Uninitialized value)", error.toString());
+  }
+  @Test
+  public void dotUnderscore() throws Exception {
+    var module = ctx.eval("enso", """
+    run op =
+      op._
+    """);
+    var run = module.invokeMember("eval_expression", "run");
+    try {
+      var error = run.execute("false_hope");
+      fail("Should never return, but: " + error);
+    } catch (PolyglotException e) {
+      assertTrue("It is exception", e.getGuestObject().isException());
+      assertEquals("Panic", e.getGuestObject().getMetaObject().getMetaSimpleName());
+      if (!e.getMessage().contains("Compiler Internal Error")) {
+        fail("Expecting Compiler Internal Error, but was: " + e.getMessage());
+      }
+    }
   }
 
   @Test

